WordPress/Classes/System/WordPressAppDelegate.swift

```diff
@@ -972,3 +972,29 @@ extension WordPressAppDelegate {
     }
 
 }
+
+// MARK: - UI Test Support
+
+extension WordPressAppDelegate {
+
+    func autoSignInUITestSite() {
+        let launchArguments = ProcessInfo.processInfo.arguments
+        guard let optIndex = launchArguments.firstIndex(of: "-ui-test-select-wpcom-site"),
+              optIndex + 1 < launchArguments.count else {
+            return
+        }
+
+        let wpComSiteAddress = launchArguments[optIndex + 1]
```

Comment:
Tip: you can use [`UserDefaults`'s ability to auto-parse dash arguments](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/UserDefaults/AboutPreferenceDomains/AboutPreferenceDomains.html#//apple_ref/doc/uid/10000059i-CH2-96930) in [its `NSArgumentDomain`](https://developer.apple.com/documentation/foundation/nsargumentdomain) here instead, which would avoid you having to parse them yourself and check the launchArguments array boundaries and all.

```suggestion
        guard let wpComSiteAddress = UserDefaults.default.string(forKey: "ui-test-select-wpcom-site") else {
            return
        }
```

Reply:
Thanks for the tip! I've used it in https://github.com/wordpress-mobile/WordPress-iOS/pull/22735/commits/70ef2531cd9f307b666430f957d3ed0c39d580bc.